src/transformation/visitors/identifier.ts

```diff
@@ -3,95 +3,114 @@ import * as lua from "../../LuaAST";
 import { transformBuiltinIdentifierExpression, checkForLuaLibType } from "../builtins";
 import { isPromiseClass, createPromiseIdentifier } from "../builtins/promise";
 import { FunctionVisitor, tempSymbolId, TransformationContext } from "../context";
-import { AnnotationKind, isForRangeType } from "../utils/annotations";
 import {
     invalidMultiFunctionUse,
-    invalidOperatorMappingUse,
     invalidRangeUse,
     invalidVarargUse,
-    invalidTableExtensionUse,
-    annotationRemoved,
+    invalidCallExtensionUse,
 } from "../utils/diagnostics";
 import { createExportedIdentifier, getSymbolExportScope } from "../utils/export";
-import { importLuaLibFeature, LuaLibFeature } from "../utils/lualib";
 import { createSafeName, hasUnsafeIdentifierName } from "../utils/safe-names";
 import { getIdentifierSymbolId } from "../utils/symbols";
-import { isMultiFunctionNode } from "./language-extensions/multi";
-import { isOperatorMapping } from "./language-extensions/operators";
-import { isRangeFunctionNode } from "./language-extensions/range";
-import { isTableExtensionIdentifier } from "./language-extensions/table";
-import { isVarargConstantNode } from "./language-extensions/vararg";
 import { isOptionalContinuation } from "./optional-chaining";
 import { isStandardLibraryType } from "../utils/typescript";
+import { getExtensionKindForNode, ExtensionKind, getExtensionKindForSymbol } from "../utils/language-extensions";
+import { callExtensions } from "./language-extensions/call-extension";
 
-export function transformIdentifier(context: TransformationContext, identifier: ts.Identifier): lua.Identifier {
-    if (isOptionalContinuation(identifier)) {
-        return lua.createIdentifier(identifier.text, undefined, tempSymbolId);
-    }
-
-    if (isMultiFunctionNode(context, identifier)) {
+function reportInvalidExtensionValue(
+    context: TransformationContext,
+    identifier: ts.Identifier,
+    extensionKind: ExtensionKind
+): void {
+    if (extensionKind === ExtensionKind.MultiFunction) {
         context.diagnostics.push(invalidMultiFunctionUse(identifier));
-        return lua.createAnonymousIdentifier(identifier);
+    } else if (extensionKind === ExtensionKind.RangeFunction) {
+        context.diagnostics.push(invalidRangeUse(identifier));
+    } else if (extensionKind === ExtensionKind.VarargConstant) {
+        context.diagnostics.push(invalidVarargUse(identifier));
     }
+}
 
-    if (isOperatorMapping(context, identifier)) {
-        context.diagnostics.push(invalidOperatorMappingUse(identifier));
-    }
+const extensionKindToValueName: { [T in ExtensionKind]?: string } = {
```

Comment:
This should probably be located somewhere in transformation/language-extensions and imported